src/components/PostDetailsSidebar.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect, useState } from 'react';
+import { Loader } from './Loader';
+import { useComments } from './Contexts/CommentsContext';
+import { Post } from '../types/Post';
+import { deleteComment, getComments } from '../api/comments';
+import { PostDetails } from './PostDetails';
+
+type Props = {
+  selectedPost: Post,
+  isNewComment: boolean,
+  onAddComment: (status: boolean) => void,
+};
+
+export const PostDetailsSidebar: React.FC<Props> = ({
+  selectedPost,
+  isNewComment,
+  onAddComment,
+}) => {
+  const { comments, setComments } = useComments();
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    if (selectedPost) {
+      getComments(selectedPost.id)
+        .then(setComments)
+        .catch(() => setHasError(true))
+        .finally(() => setIsLoading(false));
+    }
```

Comment:
```suggestion
    if (selectedPost) {
      setIsLoading(true);
    
      getComments(selectedPost.id)
        .then(setComments)
        .catch(() => setHasError(true))
        .finally(() => setIsLoading(false));
    }
```
If `selectedPost` is `null` the `isLoading` state doesn't change to false.

But, as I see `selectedPost` is a required prop, so you can get rid of ` if (selectedPost)` check